File: backend/src/main/java/hospital_management/backend/service/DoctorService.java
package hospital_management.backend.service;

import hospital_management.backend.dto.DoctorRequest;
import hospital_management.backend.entity.Doctor;

import java.util.List;

public interface DoctorService {

    // Add doctor + create login account
    Doctor addDoctor(DoctorRequest request);

    List<Doctor> getAllDoctors();

    Doctor getDoctorById(Long id);

    Doctor updateDoctor(Long id, Doctor doctor);

    void deleteDoctor(Long id);
}

File: frontend/src/services/doctorService.js
import api from "./api";

export const getDoctors = () => {
    return api.get("/doctors");
};

export const addDoctor = (doctor) => {
    return api.post("/doctors", doctor);
};

export const updateDoctor = (id, doctor) => {
    return api.put(`/doctors/${id}`, doctor);
};

export const deleteDoctor = (id) => {
    return api.delete(`/doctors/${id}`);
};